infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -44,95 +68,231 @@ public class LogReplicationConfigManager {
+                    .forEachOrdered(group -> {
+                        logicalGroupToStreams.put(group, new HashSet<>());
+                    });
+            log.debug("Targeted logical groups={}", logicalGroupToStreams.keySet());
+        }
 
         registryTableEntries.forEach(entry -> {
-            String tableName = getFullyQualifiedTableName(entry.getKey());
+            String tableName = TableRegistry.getFullyQualifiedTableName(entry.getKey());
+            UUID streamId = CorfuRuntime.getStreamID(tableName);
 
-            if (entry.getValue().getMetadata().getTableOptions().getIsFederated()) {
+            // Add merge only streams first
+            if (MERGE_ONLY_STREAMS.contains(streamId)) {
                 streamsToReplicate.add(tableName);
+                // Collect tags for merge-only stream
+                streamToTagsMap.put(streamId, Collections.singletonList(LOG_REPLICATOR_STREAM_INFO.getStreamId()));
+            }
 
-                // Collect tags for this stream
-                UUID streamId = CorfuRuntime.getStreamID(tableName);
-                List<UUID> tags = streamToTagsMap.getOrDefault(streamId, new ArrayList<>());
-                tags.addAll(entry
-                    .getValue()
-                    .getMetadata()
-                    .getTableOptions()
-                    .getStreamTagList()
-                    .stream()
-                    .map(streamTag -> TableRegistry.getStreamIdForStreamTag(entry.getKey().getNamespace(), streamTag))
-                    .collect(Collectors.toList()));
-                streamToTagsMap.put(streamId, tags);
-            } else {
-                // Registry and ProtobufDescriptor tables do not have the is_federated flag but are to be replicated.
-                if (!tableName.equals(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME)) &&
-                    !tableName.equals(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, PROTOBUF_DESCRIPTOR_TABLE_NAME))) {
-                    streamsToDrop.add(CorfuRuntime.getStreamID(tableName));
+            // Find streams to replicate for every logical group for this session
+            if (entry.getValue().getMetadata().getTableOptions().hasReplicationGroup()) {
+                String clientName = entry.getValue().getMetadata().getTableOptions().getReplicationGroup().getClientName();
+                String logicalGroup = entry.getValue().getMetadata().getTableOptions().getReplicationGroup().getLogicalGroup();
+                // TODO (V2): Client name should be checked after the rpc stream is added for Sink side session creation.
+//                if (session.getSubscriber().getClientName().equals(clientName) && groups.contains(logicalGroup)) {
+                if (isSink || logicalGroupToStreams.containsKey(logicalGroup)) {
+                    streamsToReplicate.add(tableName);
+                    Set<String> relatedStreams = logicalGroupToStreams.getOrDefault(logicalGroup, new HashSet<>());
```

Comment:
if this is a sink node, then won't this getOrDefault be executed even if the logicalGroupToStreams already has the logical group? Or do we ensure that on sink the logicalGroupToStreams will not carry the logical group?